src/parquet/arrow/reader.cc

```diff
@@ -179,7 +183,13 @@ class FileReader::Impl {
   virtual ~Impl() {}
 
   Status GetColumn(int i, std::unique_ptr<ColumnReader>* out);
+  Status ReadSchemaField(int i, std::shared_ptr<Array>* out);
+  Status ReadSchemaField(int i, const std::vector<int>& indices,
```

Comment:
I'm 100% with you on the definitions.
Also the naming make sense to me however I think the coherency and the docs are the most crucial part so I don't have as strong feelings about it :)